### PR TITLE
make pip install <root_dir> and cmake installation equivalent. make a…

### PR DESCRIPTION
**Pull request by koparasy**
Add option to disable amslib C++ installation.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ AMSTrain = "ams_wf.AMSTrain:main"
 "Homepage" = "https://github.com/LLNL/AMS/"
 
 [tool.setuptools]
+package-dir = {"" = "src/AMSWorkflow"}
```
Comment (lpottier) [inline pyproject.toml:33]: I am not sure I understand this line?

**Review by lpottier — APPROVED** [review]

**Comment by lpottier** [comment]
PR looks good, thanks.

File: pyproject.toml (hunk 1 above)
Comment (koparasy) [reply pyproject.toml:33]: We used to do something unconventional. We had the `pyproject.toml` on the project root directory. During cmake installation, we copied the `pyproject.toml` to the` build/src/AMSWorkflow/ `and then issued the `pip install` command. 

With that previous setup you could not execute `pip install` outside of `cmake` and have it working. You had to manually copy` pyproject.toml` to `src/AMSWorkflow/ `and do the install.

So this change, lets the installation script know that the directories of this package (ams) are not in the current directory but to the ones pointed by the dictionary.

The benefits are that this `git+ssh://git@github.com:LLNL/AMS.git@develop` should work once it lands. Also you can do `pip install -e ./` from the AMS root directory and you can install in develop mode the python AMS end. Completely avoiding the `cmake` path.

Does this resolve the conversation?

Comment (lpottier) [reply pyproject.toml:33]: Okay I see, it makes sense. Thanks for the detailed explanation.

**Pull request merged**